hack/generator/pkg/codegen/pipeline_convert_allof_and_oneof_to_objects.go

```diff
@@ -118,63 +122,70 @@ type synthesizer struct {
 	defs         astmodel.Types
 }
 
-func (s synthesizer) oneOfObject(oneOf astmodel.OneOfType) (astmodel.Type, error) {
-	// If there's more than one option, synthesize a type.
-	// Note that this is required because Kubernetes CRDs do not support OneOf the same way
-	// OpenAPI does, see https://github.com/Azure/k8s-infra/issues/71
-	var properties []*astmodel.PropertyDefinition
-
-	propertyDescription := "Mutually exclusive with all other properties"
-	err := oneOf.Types().ForEachError(func(t astmodel.Type, i int) error {
-		prop, err := s.convertToOneOfProperty(i, t)
-		if err != nil {
-			return err
-		}
+type names struct {
+	golang astmodel.PropertyName
+	json   string
 
-		prop = prop.MakeOptional()
-		prop = prop.WithDescription(propertyDescription)
+	// used to resolve conflicts:
+	goodName bool
+	depth    int
+}
 
-		properties = append(properties, prop)
-		return nil
-	})
+func (ns names) betterThan(other names) bool {
+	if ns.goodName && !other.goodName {
+		return true
+	}
 
-	if err != nil {
-		return nil, err
+	if other.goodName && !ns.goodName {
```

Comment:
I think this would be clearer (easier to compare both conditions) if the parameters were in the same order as above:
```suggestion
	if !ns.goodName && other.goodName {
```